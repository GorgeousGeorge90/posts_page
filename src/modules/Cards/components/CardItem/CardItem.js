import {Card} from 'react-bootstrap';

const CardItem = ({description,title,link,pic}) => {

    return (<li>
        <Card style={{width: '18rem'}}>
            <Card.Img variant="top"
                      src={pic}
            />
            <Card.Body>
                <Card.Title>{title}</Card.Title>
                <Card.Text>
                    {description}
                </Card.Text>
                <Card.Link href={link}>{link}</Card.Link>
            </Card.Body>
        </Card>
    </li>)
}

export default CardItem