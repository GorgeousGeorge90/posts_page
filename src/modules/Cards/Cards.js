import styles from './Cards.module.scss';
import {useSelector} from "react-redux";
import CardItem from "./components/CardItem/CardItem";
import {getIsLoading, getWorks} from "./selectors/selectors";
import Preloader from "../../UI/Preloader/Preloader";


const Cards = () => {
    const works = useSelector(getWorks)
    const isLoading = useSelector(getIsLoading)

    if (isLoading) {
        return <Preloader/>
    }

    return (<div className={styles.container}>
        <ul>
            {
                works.map(work =><CardItem key={work.id}
                                                {...work}
                />)
            }
        </ul>
        </div>)
}

export default Cards