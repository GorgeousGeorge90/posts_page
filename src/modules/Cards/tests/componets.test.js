import {useSelector} from 'react-redux';
import {render,screen} from '@testing-library/react';
import CardItem from "../components/CardItem/CardItem";
import Cards from "../Cards";

jest.mock('react-redux')

describe('test Cards components',()=> {
    it('goal: render CardItem component',()=> {
        const data = {
            description:'Hi',
            title:'Title',
            link:'nice',
            pic:'pic',
        }

        useSelector.mockReturnValue(data)
        const component = render(<CardItem {...data}/>)
        const cardItem = screen.getByRole('listitem')
        expect(component).toMatchSnapshot()
        expect(cardItem).toHaveTextContent('Title')
    })

    it('goal: render Cards component',()=> {
        const data = {
            works:[
                {id:1, title: 'Joke', body:'Lol'}
            ],
            isLoading: false,
        }
        useSelector.mockReturnValue(data)
        const component = render(<Cards {...data}/>)
        expect(component).toMatchSnapshot()
    })
})