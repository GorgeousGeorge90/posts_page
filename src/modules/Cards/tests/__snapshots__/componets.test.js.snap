// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Cards components goal: render CardItem component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li>
        <div
          class="card"
          style="width: 18rem;"
        >
          <img
            class="card-img-top"
            src="pic"
          />
          <div
            class="card-body"
          >
            <div
              class="card-title h5"
            >
              Title
            </div>
            <p
              class="card-text"
            >
              Hi
            </p>
            <a
              class="card-link"
              href="nice"
            >
              nice
            </a>
          </div>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li>
      <div
        class="card"
        style="width: 18rem;"
      >
        <img
          class="card-img-top"
          src="pic"
        />
        <div
          class="card-body"
        >
          <div
            class="card-title h5"
          >
            Title
          </div>
          <p
            class="card-text"
          >
            Hi
          </p>
          <a
            class="card-link"
            href="nice"
          >
            nice
          </a>
        </div>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test Cards components goal: render Cards component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="loader"
      >
        <div
          class="spinner-border"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="loader"
    >
      <div
        class="spinner-border"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
